spark-connector/spark-connector/build.gradle.kts

```diff
@@ -40,3 +41,26 @@ dependencies {
   testImplementation("org.scala-lang.modules:scala-java8-compat_$scalaVersion:$scalaJava8CompatVersion")
   testRuntimeOnly(libs.junit.jupiter.engine)
 }
+
+sourceSets {
+  main {
+    scala {
+      setSrcDirs(listOf("src/main/scala", "src/main/java"))
+    }
+    java {
+      setSrcDirs(emptyList<String>())
+    }
+  }
+  test {
+    scala {
+      setSrcDirs(listOf("src/test/scala", "src/test/java"))
+    }
+    java {
+      setSrcDirs(emptyList<String>())
+    }
+  }
+}
+
+tasks.withType(ScalaCompile::class.java).configureEach {
+  scalaCompileOptions.additionalParameters = listOf("-target:jvm-1.8")
```

Comment:
https://github.com/gradle/gradle/issues/19456

Reply:
> [gradle/gradle#19456](https://github.com/gradle/gradle/issues/19456)

without this, we will get the error, `bad option: '-target:jvm-1.11'` or `bad option: '-target:jvm-1.17'`, seems the only solution would be to force the jvm 8 compatibility.

Reply:
> [gradle/gradle#19456 (comment)](https://github.com/gradle/gradle/issues/19456#issuecomment-1004718359)

without this, we will get the error, `bad option: '-target:jvm-1.11'` or `bad option: '-target:jvm-1.17'`, seems the only solution would be to force the jvm 8 compatibility.